pages/glossary/programmatically-determined-link-context.md

```diff
@@ -12,6 +12,7 @@ The _programmatically determined context_ of a link (or _programmatically determ
 
 - being an [ancestor][] of the link in the [flat tree][] with a [semantic role][] of `listitem`; or
 - being the closest [ancestor][] of the link in the [flat tree][] that is a `p` element; or
+- being the [parent][] of the link in the [flat tree][] if it contains [text nodes][] that are not contained by the link; or
```

Comment:
Currently, the rule passes
```html
<p><a href="#lorem">Read more</a> about Lorem</p>
<p><a href="#ipsum">Read more</a> about Ipsum</p>
```
but fails
```html
<div><a href="#lorem">Read more</a> about Lorem</div>
<div><a href="#ipsum">Read more</a> about Ipsum</div>
```

I'm having a hard time to believe that the later violate 2.4.4 (especially if we say that the former satisfies it).

It is true that H77, H78, H79, H81 list some elements (and there is also a [WCAG def of programatically determined link context](https://www.w3.org/TR/WCAG21/#dfn-programmatically-determined-link-context), but they are also all englobed into [G53: Identifying the purpose of a link using link text combined with the text of the enclosing sentence](https://www.w3.org/WAI/WCAG21/Techniques/general/G53). I would argue that the later example passes G53.

Reply:
Using @Jym77's suggestion of updating this to use display:block
I believe this to be a better match for the semantics of paragraph in the WCAG definition of programatically determined link context.

Reply:
Again I should say, why is it any block? I think this is somewhat speculative, and may be going outside of the intent of the SC. I think it would be good to see if AG could weigh in on this. I've certainly never tested it how this is being suggested, and would fail things that this rule would now pass.

Reply:
That would be false negatives for the rule, which we are sort of okay with, right? 
But considering the examples from Jean-Yves, wouldn't you pass some examples that the rule is currently failing?